src/hd.rs

```diff
@@ -375,33 +308,33 @@ fn batch_import(rpc: &RpcClient, import_reqs: Vec<(Address, RescanSince, String)
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum KeyOrigin {
-    Derived(Fingerprint, u32),
+    Derived(DescriptorChecksum, u32),
     Standalone,
     // bwt never does hardended derivation itself, but can receive an hardend --bare-xpub
-    DerivedHard(Fingerprint, u32),
+    DerivedHard(DescriptorChecksum, u32),
 }
 
 impl_string_serializer!(
     KeyOrigin,
     origin,
     match origin {
         KeyOrigin::Standalone => "standalone".into(),
-        KeyOrigin::Derived(parent_fingerprint, index) => {
-            format!("{}/{}", parent_fingerprint, index)
+        KeyOrigin::Derived(parent_desc_check, index) => {
+            format!("{}/{}", parent_desc_check, index)
```

Comment:
Need to check if there's some other customary notation for referring to the nth script of a descriptor identified by its checksum (a quick search didn't turn up anything).